tls/s2n_key_update.c

```diff
@@ -31,6 +31,7 @@ int s2n_check_record_limit(struct s2n_connection *conn, struct s2n_blob *sequenc
 int s2n_key_update_recv(struct s2n_connection *conn, struct s2n_stuffer *request)
 {
     notnull_check(conn);
+    ENSURE_POSIX(!conn->config->quic_enabled, S2N_ERR_BAD_MESSAGE);
```

Comment:
Why doesn't this one get the S2N_ERR_UNSUPPORTED_WITH_QUIC?

Reply:
S2N has different types / categories of errors.

S2N_ERR_UNSUPPORTED_WITH_QUIC is a usage error. That generally means it indicates that the customer / caller did something wrong, like called an API with illegal arguments or configure something incorrectly.

S2N_ERR_BAD_MESSAGE is a protocol error. That generally means that something unexpected / not allowed by the protocol specification happened, like the wrong handshake message was received.

Receiving a key update message from a peer while using QUIC is a protocol error, not a usage error.

Reply:
Thanks for the explanation